inference-engine/tests/functional/plugin/cpu/single_layer_tests/space_to_depth.cpp

```diff
@@ -2,194 +2,272 @@
+        for (const auto& item : shapes.second) {
+            results << CommonTestUtils::vec2str(item) << "_";
+        }
+        results << "Prc=" << inType << "_";
+        switch (mode) {
+            case SpaceToDepth::SpaceToDepthMode::BLOCKS_FIRST:
+                results << "BLOCKS_FIRST_";
+                break;
+            case SpaceToDepth::SpaceToDepthMode::DEPTH_FIRST:
+                results << "DEPTH_FIRST_";
+                break;
+            default:
+                throw std::runtime_error("Unsupported SpaceToDepthMode");
+        }
+        results << "BS=" << blockSize << "_";
+        results << CPUTestsBase::getTestCaseName(cpuParams);
+
+        return results.str();
     }
 protected:
     void SetUp() override {
-        LayerTestsDefinitions::spaceToDepthParamsTuple basicParamsSet;
+        InputShape shapes;
+        ElementType inType;
+        SpaceToDepth::SpaceToDepthMode mode;
+        std::size_t blockSize;
         CPUSpecificParams cpuParams;
-        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(shapes, inType, mode, blockSize, cpuParams) = this->GetParam();
 
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
-
-        std::vector<size_t> inputShape;
-        SpaceToDepth::SpaceToDepthMode mode;
-        std::size_t blockSize;
-        InferenceEngine::Precision netPrecision;
-        std::tie(inputShape, netPrecision, mode, blockSize, targetDevice) = basicParamsSet;
-
-        inPrc = outPrc = netPrecision;
-        selectedType = getPrimitiveType() + "_" + inPrc.name();
-        auto inPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
-        auto params = ngraph::builder::makeParams(inPrc, {inputShape});
-        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
-        auto d2s = ngraph::builder::makeSpaceToDepth(paramOuts[0], mode, blockSize);
-        d2s->get_rt_info() = getCPUInfo();
-        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(d2s)};
-        function = std::make_shared<ngraph::Function>(results, params, "SpaceToDepth");
+        if (selectedType.empty()) {
+            selectedType = getPrimitiveType();
+        }
+        selectedType = selectedType + "_" + InferenceEngine::details::convertPrecision(inType).name();
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        init_input_shapes({shapes});
+
+        auto params = ngraph::builder::makeDynamicParams(inType, inputDynamicShapes);
+        auto d2s = ngraph::builder::makeSpaceToDepth(params[0], mode, blockSize);
+        function = makeNgraphFunction(inType, params, d2s, "SpaceToDepthCPU");
     }
 };
 
 TEST_P(SpaceToDepthLayerCPUTest, CompareWithRefs) {
     SKIP_IF_CURRENT_TEST_IS_DISABLED()
 
-    Run();
-    CheckPluginRelatedResults(executableNetwork, "SpaceToDepth");
+    run();
+    // TODO: need to uncomment when this method will be updated
```

Comment:
could you uncomment please?

Reply:
Sorry, I thought I have applied the changes. But I applied changes in another file *depth_to_space.cpp*. I have reverted this depth_to_space changes and applied it into space_to_depth.cpp.